File: regiondeployer/DirectoryZipper.cs
using System.IO;
using System.IO.Compression;

namespace regiondeployer
{
    public class ZipManager
    {
        public const string DefaultZipName = "regionfarmer";

        public string ZipUpProject(string projectPath, string outputFileName, Action<string> logMessage)
        {
            logMessage($"Zipping project directory... {projectPath}");
            string output = Path.Combine(Directory.GetCurrentDirectory(), $"{outputFileName}.zip");

            if (File.Exists(output))
            {
                File.Delete(output);
            }

            ZipFile.CreateFromDirectory(projectPath, output);
            logMessage("directory zipped.");
            return output;
        }
    }
}

File: regiondeployer/directoryzipper.fs
module regiondeployer.directoryzipper

open System.IO.Compression
open System.IO

[<Literal>]
let public defaultzipname = "regionfarmer"    

///returns the named address of the output zip file
let ZipUpProject (projectpath:string, outputfilename:string, logmessage:string -> unit) : string =
    logmessage ("Zipping project directory... " + projectpath)
    let output = (Directory.GetCurrentDirectory() + "\\" + outputfilename + ".zip") //destination must be different from the source. otherwise, zipfile.createfromdirectory() wont work!

    if File.Exists output then
        File.Delete output

    ZipFile.CreateFromDirectory(projectpath, output)
    logmessage "directory zipped."
    output
